### ISSUE TITLE
Not all Puppet <4 messages are parsed

### ISSUE THREAD

**Issue by rski** [flycheck/flycheck#995]
This is similar to #675, but I wasn't sure if I should bump a year-old issue so I opted for opening a new one. 
Similarly to #675, some syntax errors are not parsed properly.
For example:
```puppet
file { 'foo':
   ensure => file
   content => 'bar'
}
```
There is a comma missing after file, which is a syntax error. Instead of it being marked, i get 
`Suspicious state from syntax checker puppet-parser: Checker puppet-parser returned non-zero exit code 1, but no errors from output: Error: Could not parse for environment production: Syntax error at content, expected '}' at line 4`
Once this error goes away there is no indication in the buffer that there is a syntax error.

```
M-x  flycheck-verify-setup
Syntax checkers for buffer test.pp in puppet-mode:

  puppet-parser
    - predicate:  t
    - executable: Found at /usr/bin/puppet

  puppet-lint
    - predicate:  t
    - executable: Found at /home/rski/bin/puppet-lint

```
Emacs 24.5.2 on CentOS 7.2
flycheck version 29snapshot (package: 20160606.302)
puppet version: 3.8.4, puppet-lint version: 1.1.0

**Comment by lunaryorn** [comment]
@rski The error pattern seems to be incorrect.  Would you like to try and fix it?

**Comment by rski** [comment]
@lunaryorn will definitely do, not just now but i'll try to make time for it soon

**Comment by rski** [comment]
Well, I can't reproduce this at my home computer with arch. Emacs versions and puppet-lint versions are the same. This is weird.

**Comment by rski** [comment]
There is another edge case of the same output which was not fixed by the PR i proposed:
```puppet
file { 'test':
  ensure => file,
  contents => something'
}
```
The unbalanced quotes cause an error like 
```
Syntax error at '
}
```
The newline causes the error not to be parsed.
This might also be an issue with the puppet 4 parser, but I'll have to check.
